GraphsV2/ProgressBarGraph/index.js

```diff
@@ -0,0 +1,282 @@
+        width * barData[usedData] * 0.01;
+}
+
+const getData = ({
+    barData,
+    maxData,
+    usedData,
+    display,
+    maxDataFormat,
+    defaultRange,
+    units
+}) => {
+
+    if (display === PERCENTAGE) {
+        const data = barData[maxData] ? parseInt(barData[usedData] * 100 / barData[maxData]) : barData[usedData];
+        return `${data}%`;
+    }
+
+    barData[usedData] = format(maxDataFormat)(barData[usedData]);
+    barData[maxData] = barData[maxData] ? format(maxDataFormat)(barData[maxData]) : undefined;
+    const dataUnits = barData['unit'] || units ? ` ${barData['unit'] || units}` : '';
+
+    return barData[maxData] ? `${barData[usedData]}${dataUnits}/ ${barData[maxData]}${dataUnits}` : `${barData[usedData]}${dataUnits}/ ${defaultRange}${dataUnits}`;
+}
+
+const getPercentage = ({
+    barData,
+    width,
+    maxData,
+    usedData
+}) => {
+    return barData[maxData] ? (width * barData[usedData] / barData[maxData]) * 100 / width : barData[usedData];
+}
+
+const getColor = ({
+    barData,
+    width,
+    barColor,
+    colorRange,
+    maxData,
+    usedData
+}) => {
+
+    let setColor = false;
+
+    if (colorRange) {
+        const percentage = getPercentage(barData, width, maxData, usedData);
+        let colorData;
+        colorRange.filter((d) => {
+            if (percentage <= d['upto'] && !setColor) {
+                setColor = true;
+                colorData = d;
+            }
+        });
+
+        return colorData['color'];
+    }
+
+    return barColor;
+}
+
+const Container = styled('div')({
+    fontSize: ({ fontSize } = {}) => fontSize,
+    width: ({ width } = {}) => width,
+    height: ({ height } = {}) => height,
+    padding: '0.6rem',
+    overflowY: 'auto',
+    overflowX: 'hidden',
+    position: 'relative',
+});
+
+const Item = styled('div')({
+    width: ({ width } = {}) => width,
+    height: ({ height } = {}) => height,
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'column',
+});
+
+const UpperText = styled('div')({
+    fontSize: ({ fontSize } = {}) => fontSize,
+    order: 1,
+    marginBottom: '-2px',
+});
+
+const BarSection = styled('div')({
+    order: 2,
+    display: 'flex',
+    flexDirection: ({ direction } = {}) => direction,
+});
+
+const InnerBarSection = styled('div')({
+    order: 1,
+    flex: 1,
+    height: ({ height } = {}) => height,
+});
+
+const LowerText = styled('div')({
+    fontSize: ({ fontSize } = {}) => fontSize,
+    order: 2,
+    flex: 'auto',
+    alignSelf: ({ alignSelf } = {}) => alignSelf,
+    marginRight: ({ marginRight } = {}) => marginRight,
+});
+
+const Tooltip = styled('div')({});
+
+const ProgressBarGraph = (props) => {
+    const [customTooltips, setCustomTooltips] = useState({});
+
+    const {
+        data,
+        width,
+        height,
+        properties
+    } = props;
+
+    useEffect(() => {
+        setTooltip();
+    }, [props.data, props.width, props.height])
+
+    const setTooltip = () => {
```

Comment:
you don't really need this function, let's remove and add the code below in useEffect.